packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -35,14 +49,39 @@ export const commerceFacetSetReducer = createReducer(
         const {facetId, selection} = action.payload;
         const facetRequest = state[facetId]?.request;
 
-        if (!facetRequest) {
+        if (!facetRequest || facetRequest.type !== 'regular') {
```

Comment:
Otherwise we're implicitly trusting that the correct toggle select action was dispatched for that facet.

If actions could only be dispatched from our controllers I suppose that would be OK. However, let's suppose the action is dispatched from custom code, nothing would prevent an implementer from dispatching a toggleSelectFacetValue on, say, a numeric facet value. If we don't return here, we might end up doing some non-sensical state updates.

Thoughts?

Reply:
No strong opinions. Shouldn't the check be made in the `validatePayload` part of the action though? 🤔

Reply:
We can't: what we want to validate is in the state, not in the payload

Reply:
Hah! Ok I'm fine with leaving it as-is.